system/ee/ExpressionEngine/Controller/Members/Roles/Roles.php

```diff
@@ -662,11 +662,24 @@ private function renderRoleTab(Role $role, $errors)
                     'fields' => [
                         'require_mfa' => [
                             'type' => 'yes_no',
+                            'disabled' => version_compare(PHP_VERSION, 7.1, '<'),
                             'value' => $role->RoleSettings->filter('site_id', ee()->config->item('site_id'))->first()->require_mfa,
                         ]
                     ]
                 ],
             ]);
+            if (version_compare(PHP_VERSION, 7.1, '<')) {
+                ee()->lang->load('addons');
+                $section = array_merge($section, [
+                    ee('CP/Alert')->makeInline('mfa_not_available')
+                        ->asWarning()
+                        ->withTitle(lang('mfa_not_available'))
+                        ->addToBody(sprintf(lang('version_required'), 'PHP', 7.1))
+                        ->cannotClose()
+                        ->render()
+                        . form_hidden('require_mfa', 'n')
+                ]);
+            }
```

Comment:
Not loving the PHP version check and a specific functionality warning being generated in line here.

What if we abstract this a bit.  Could we make this part of the MFA service?  (Maybe both the addition of the setting as well as the check and inline warning) ?

Tagging @intoeetive and @bryannielsen for additional thoughts?

In part I'm thinking this is the first minor release of Pro... it's likely we'll run into this multiple times in the next few years.